src/library/scala/io/BufferedSource.scala

```diff
@@ -83,17 +84,21 @@ class BufferedSource(inputStream: InputStream, bufferSize: Int)(implicit val cod
 
   override def getLines(): Iterator[String] = new BufferedLineIterator
 
-  /** Efficiently converts the entire remaining input into a string. */
-  override def mkString = {
-    // Speed up slurping of whole data set in the simplest cases.
-    val allReader = decachedReader
-    val sb = new StringBuilder
-    val buf = new Array[Char](bufferSize)
-    var n = 0
-    while (n != -1) {
-      n = allReader.read(buf)
-      if (n>0) sb.appendAll(buf, 0, n)
+  /** Efficiently appends the entire remaining input. */
+  override def addString(sb: StringBuilder, start: String, sep: String, end: String): StringBuilder =
+    if (sep.length != 0) mkString.addString(sb, start, sep, end)
```

Comment:
The `sep != ""` case expands to:
```scala
addString(new StringBuilder, "", "", "").toString.addString(sb, start, sep, end)
```
It uses the `""` case to convert to a String, and then passes off to StringOps for interspersing sep.